beta/examples/tensorflow/classification/main.py

```diff
@@ -115,12 +115,26 @@ def load_checkpoint(model, ckpt_path):
     return None
 
 
-def resume_from_checkpoint(model, compression_ctrl, ckpt_path, steps_per_epoch):
+def get_scheduler_state(num_steps, steps_per_epoch, config):
+    current_step = num_steps - 1
+    current_epoch = current_step // steps_per_epoch
+    scheduler_state = {'current_step': current_step, 'current_epoch': current_epoch}
+
+    if isinstance(config.compression, list) and len(config.compression) > 1:
+        scheduler_state = [scheduler_state for _ in config.compression]
+
+    return scheduler_state
+
+
+def resume_from_checkpoint(model, compression_ctrl, ckpt_path, steps_per_epoch, config):
     if load_checkpoint(model, ckpt_path) == 0:
         return 0
     initial_step = model.optimizer.iterations.numpy()
     initial_epoch = initial_step // steps_per_epoch
-    compression_ctrl.scheduler.load_state(initial_step, steps_per_epoch)
+
+    scheduler_state = get_scheduler_state(initial_step, steps_per_epoch, config)
```

Comment:
do you really need to calculate parameters for getting scheduler state? why not just load them from the checkpoint in a resume mode?

Reply:
Thanks, it is a good idea. I create a separate issue (CVS-49551) to implement the saving and loading scheduler's state from the checkpoint. It will be implemented as separate PR.